src/averageAges.js

```diff
@@ -15,11 +15,23 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const manOnly = people.filter(person => person.sex === 'm');
+  let sumManAge;
+
+  if (!century) {
+    sumManAge = manOnly.reduce((sumAge, person) =>
+      sumAge + (person.died - person.born), 0);
```

Comment:
You count the total number of ages several times for the entire code, this functionality can be transferred to a separate function and used as a helper.